conans/client/build/compiler_flags.py

```diff
@@ -47,8 +47,8 @@ def architecture_flag(settings):
         return ""
 
     if str(compiler) in ['gcc', 'apple-clang', 'clang', 'sun-cc']:
-        if str(the_os) == 'Macos' and str(subsystem) == 'Catalyst' and str(arch) == 'x86_64':
-            return '-target=x86_64-apple-ios-macabi'
+        if str(the_os) == 'Macos' and str(subsystem) == 'catalyst' and str(arch) == 'x86_64':
+            return '--target=x86_64-apple-ios-macabi'
```

Comment:
I know this is not from this PR, but I think this --target might conflict with the --target that the AutotoolsBuildEnvironment will add. Please have a look at it.

Reply:
I don't think so, this one is a compiler flag, while AutoToolsBuildEnvironment only passes flag to the configure script itself

Reply:
I am not sure, this is being added in AutoToolsBuildEnvironment as flags:
```python
 def _configure_flags(self):
        ret = list(self._deps_cpp_info.cflags)
        arch_flag = architecture_flag(self._conanfile.settings)
        if arch_flag:
            ret.append(arch_flag)
```
Then ``configure()`` itself is also adding:
```python
 if target is not False:  # Skipped by user
            if target or self.target:  # User specified value or automatic
                triplet_args.append("--target=%s" % (target or self.target))
```

Reply:
I can see that this is difficult to reproduce, as getting the --target is not easy unless building a graph with build_requires that will add the --target triplet, but still looks possible. Maybe we want to move forward and leave this as an open issue for later.

Reply:
I see the `--target` from @SSE4 changes affecting the `configure` command-line call, while the existing `--target` could be added to the _flags_ that are populating the environment (hard to follow having a look to the sources). It looks like both can collide...

Reply:
Yes exactly. The only thing is that at the moment this is very unlikely, it will only happen when the --target is involved (not many times, only special build context builds) and the new catalyst. 

We are already starting to decouple this helper into 3 of them, I think we could figure this out in the new ones.

Reply:
I'm totally ok to forget about this issue now... and focus on how this information will be propagated from build-requires, or how to mix it when coming from different sources.